src/components/article/Article.jsx

```diff
@@ -6,8 +6,8 @@ import { NewsDataContext } from "../../data/NewData";
 
 const Article = ({ article, userList }) => {
     const [userAvatar, setUserAvatar] = useState("");
-
-    const { setArticleForComment } = useContext(NewsDataContext);
+    const [isMore, setIsMore] = useState(false);
+    const { setActiveArticle } = useContext(NewsDataContext);
```

Comment:
As discussed, you should not be getting the selected article in this way as this could lead to stale data.